standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java

```diff
@@ -4059,7 +3806,7 @@ public Partition append_partition_with_environment_context(final String dbName,
     String location;
 
     PartValEqWrapperLite(Partition partition) {
-      this.values = partition.isSetValues()? partition.getValues() : null;
+      this.values = partition.isSetValues()? new ArrayList<>(partition.getValues()) : null;
```

Comment:
is this a bugfix? if yes - could you open a separate jira to submit it?

Reply:
Not a bug really,  just to clean up the codes.  The following pieces of codes can be simply treated as `lhsValues.equals(rhsValues)` by `equals` of an ArrayList.

     if (lhsValues.size() != rhsValues.size()) {
        return false;
       }

      for (int i=0; i<lhsValues.size(); ++i) {
        String lhsValue = lhsValues.get(i);
        String rhsValue = rhsValues.get(i);

        if ((lhsValue == null && rhsValue != null)
            || (lhsValue != null && !lhsValue.equals(rhsValue))) {
          return false;
        }
      }

      return true;`